tests/operators/test_python.py

```diff
@@ -1222,23 +1230,42 @@ def f(templates_dict):
         self._run_as_operator(f, templates_dict={'ds': '{{ ds }}'})
 
     def test_airflow_context(self):
-        def f(**context):
-            # not serializable
-            assert 'var' not in context
-            assert 'task_instance' not in context
-            assert 'ti' not in context
-            # require airflow
-            assert 'conf' in context
-            assert 'dag' in context
-            assert 'dag_run' in context
-            assert 'macros' in context
-            assert 'task' in context
-            # require airflow or pendulum
-            assert 'execution_date' in context
-            assert 'next_execution_date' in context
-            assert 'prev_execution_date' in context
-            assert 'prev_execution_date_success' in context
-            assert 'prev_start_date_success' in context
+        def f(
+            # basic
+            ds_nodash,
+            inlets,
+            next_ds,
+            next_ds_nodash,
+            outlets,
+            params,
+            prev_ds,
+            prev_ds_nodash,
+            run_id,
+            task_instance_key_str,
+            test_mode,
+            tomorrow_ds,
+            tomorrow_ds_nodash,
+            ts,
+            ts_nodash,
+            ts_nodash_with_tz,
+            yesterday_ds,
+            yesterday_ds_nodash,
+            # pendulum-specific
+            execution_date,
+            next_execution_date,
+            prev_execution_date,
+            prev_execution_date_success,
+            prev_start_date_success,
+            # airflow-specific
+            macros,
+            conf,
+            dag,
+            dag_run,
+            task,
+            # other
+            **context
+        ):  # pylint: disable=unused-argument,too-many-arguments,too-many-locals
+            pass
```

Comment:
@turbaszek what do you think of this? I am testing it by forcing to unpack the keys given.

Reply:
Also if there isn't anymore I could do I would love to get this merged. :)

Reply:
> I am testing it by forcing to unpack the keys given.

More like a system-test instead of unit-test now.

Reply:
Hm, would it be possible to use paramaterization of tetst?

Reply:
Good point, but I think this would rather make it more complex.. of course we could save a few lines but.. as it is right now it is pretty straight forward to see what is going on. Using `parameterized.expand` would make it more complex which is not worth the lines we would save in my opinion.

Reply:
Actually i think it isn't even **possible** because the function is executed in a different context so I cannot pass variables/parameters from outside into it.

Reply:
What would work would be using jinja to create the tests :D but I think that is really too much.